packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -468,8 +468,16 @@ Static group authorization should perform as expected.`,
   ): void => {
     const resolver = ctx.resolvers.getResolver(typeName, fieldName) as TransformerResolverProvider;
     const roleDefinitions = acm.getRolesPerOperation('read').map(r => this.roleMap.get(r)!);
-    const primaryFields = getTable(ctx, def).keySchema.map(att => att.attributeName);
-    const authExpression = generateAuthExpressionForQueries(this.configuredAuthProviders, roleDefinitions, def.fields ?? [], primaryFields);
+    const tableKeySchema = getTable(ctx, def).keySchema;
+    const primaryFields = tableKeySchema.map(att => att.attributeName);
+    const primaryKey = tableKeySchema.find((att: any) => att.keyType === 'HASH')!.attributeName;
```

Comment:
Could you please move this expression into a function called getPrimaryKey or getPartitionKey.

Reply:
added